src/ResourceManagement/DataLake.Store/Microsoft.Azure.Management.DataLake.Store/Customizations/DataTransfer/DataLakeStoreTransferClient.cs

```diff
@@ -529,6 +539,50 @@ private UploadMetadata GetMetadata()
             }
         }
 
+        private int GetDefaultPerFileThreadCount(long fileLength, long maxSegmentLength)
+        {
+            return (int)Math.Min(Math.Ceiling((double)fileLength / maxSegmentLength), DefaultMaxPerFileThreadCount);
+        }
+
+        private int GetDefaultPerFileThreadCountForFolder(long folderSize, long fileCount, long maxSegmentLength, out int concurrentFileCount)
+        {
+            if (this.Parameters.ConcurrentFileCount > 0)
+            {
+                concurrentFileCount = this.Parameters.ConcurrentFileCount;
+            }
+            else
+            {
+                concurrentFileCount = DefaultIdealConcurrentFileCount;
+            }
+
+            long avgFileSize = folderSize / fileCount;
```

Comment:
agreed and fixed, we now throw if there are no files, since we can't upload nothing :)